test/test_autograd.py

```diff
@@ -6188,6 +6192,258 @@ def foo(a):
         self.assertEqual(vhp, torch.mm(v.unsqueeze(0), hes).squeeze(0))
 
 
+class TestAutogradForwardMode(TestCase):
+    def tearDown(self):
+        # Ensure that a failing test won't make others fail
+        while fwAD._current_level >= 0:
+            fwAD.exit_dual_level()
```

Comment:
This is what context managers are for right

Reply:
Yes but some tests here test the non-context manager API and so could lead to issues.